### PR TITLE
benchmarks: don't block shutdown when using netty epoll

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java [hunk 1]
```diff
@@ -109,10 +112,11 @@ static Server newServer(ServerConfiguration config) throws IOException {
     final EventLoopGroup boss;
     final EventLoopGroup worker;
     final Class<? extends ServerChannel> channelType;
+    ThreadFactory tf = new DefaultThreadFactory("server-elg-", true /*daemon */);
     switch (config.transport) {
       case NETTY_NIO: {
-        boss = new NioEventLoopGroup();
-        worker = new NioEventLoopGroup();
+        boss = new NioEventLoopGroup(0, tf);
```
Comment (ejona86) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java]: The bosses should probably just have one thread.

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java [hunk 2]
```diff
@@ -82,6 +84,7 @@ public void run() {
         }
       }
     });
+    server.awaitTermination();
```
Comment (ejona86) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java]: You can delete the awaitTermination in the shutdownHook.

Comment (carl-mastrangelo) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java]: Done.

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java (hunk 1 above)
Comment (carl-mastrangelo) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java]: Done

**Comment by carl-mastrangelo** [comment]
I'll update the desc when I squash

**Pull request merged**